core/src/main/java/io/questdb/griffin/engine/functions/date/ExtractFromTimestampFunctionFactory.java

```diff
@@ -0,0 +1,423 @@
+
+    @Override
+    public String getSignature() {
+        return "extract(sN)";
+    }
+
+    @Override
+    public Function newInstance(
+            int position,
+            ObjList<Function> args,
+            IntList argPositions,
+            CairoConfiguration configuration,
+            SqlExecutionContext sqlExecutionContext
+    ) throws SqlException {
+        final CharSequence part = args.getQuick(0).getStr(null);
+        final Function arg = args.getQuick(1);
+
+        if (SqlKeywords.isCenturyKeyword(part)) {
+            return new CenturyFunction(arg);
+        }
+
+        if (SqlKeywords.isDayKeyword(part)) {
+            return new DayOfMonthFunctionFactory.DayOfMonthFunction(arg);
+        }
+
+        if (SqlKeywords.isDecadeKeyword(part)) {
+            return new DecadeFunction(arg);
+        }
+
+        if (SqlKeywords.isDowKeyword(part)) {
+            return new DowFunction(arg);
+        }
+
+        if (SqlKeywords.isDoyKeyword(part)) {
+            return new DoyFunction(arg);
+        }
+
+        if (SqlKeywords.isEpochKeyword(part)) {
+            return new EpochFunction(arg);
+        }
+
+        if (SqlKeywords.isHourKeyword(part)) {
+            return new HourOfDayFunctionFactory.HourOfDayFunction(arg);
+        }
+
+        if (SqlKeywords.isIsoDowKeyword(part)) {
+            return new IsoDowFunction(arg);
+        }
+
+        if (SqlKeywords.isIsoYearKeyword(part)) {
+            return new IsoYearFunction(arg);
+        }
+
+        if (SqlKeywords.isMicrosecondsKeyword(part)) {
+            return new MicrosecondsFunction(arg);
+        }
+
+        if (SqlKeywords.isMillenniumKeyword(part)) {
+            return new MillenniumFunction(arg);
+        }
+
+        if (SqlKeywords.isMillisecondsKeyword(part)) {
+            return new MillisecondsFunction(arg);
+        }
+
+        if (SqlKeywords.isMinuteKeyword(part)) {
+            return new MinuteOfHourFunctionFactory.MinuteFunction(arg);
+        }
+
+        if (SqlKeywords.isMonthKeyword(part)) {
+            return new MonthOfYearFunctionFactory.MonthOfYearFunction(arg);
+        }
+
+        if (SqlKeywords.isQuarterKeyword(part)) {
+            return new QuarterFunction(arg);
+        }
+
+        if (SqlKeywords.isSecondKeyword(part)) {
+            return new SecondOfMinuteFunctionFactory.SecondOfMinuteFunc(arg);
+        }
+
+        if (SqlKeywords.isWeekKeyword(part)) {
+            return new WeekFunction(arg);
+        }
+
+        if (SqlKeywords.isYearKeyword(part)) {
+            return new YearFunctionFactory.YearFunction(arg);
+        }
+
+        throw SqlException.position(argPositions.getQuick(0)).put("unsupported part '").put(part).put('\'');
+    }
+
+    static final class CenturyFunction extends IntFunction implements UnaryFunction {
+
+        private final Function arg;
+
+        public CenturyFunction(Function arg) {
+            super();
```

Comment:
all these calls to the parent class' constructor are unnecessary, for all inner static classes.

Reply:
removed